CMakeLists.txt

```diff
@@ -2,92 +2,114 @@ cmake_minimum_required(VERSION 3.15)
+  set(PROJECT_VERSION "${PROJECT_VERSION}-${PROJECT_VERSION_FLAVOUR}")
+  add_compile_definitions(FW_FLAVOR=${PROJECT_VERSION_FLAVOUR})
+  if(NOT PROJECT_VERSION_FLAVOUR_REVISION STREQUAL "")
+    set(PROJECT_VERSION "${PROJECT_VERSION}${PROJECT_VERSION_FLAVOUR_REVISION}")
+    add_compile_definitions(FW_FLAVERSION=${PROJECT_VERSION_FLAVOUR_REVISION})
+  endif()
+endif()
 
 # Inform user about the resolved settings
-MESSAGE(STATUS "Project version: ${PROJECT_VERSION}")
-MESSAGE(
+message(STATUS "Project version: ${PROJECT_VERSION}")
+message(
   STATUS "Project version with short suffix: ${PROJECT_VERSION}${PROJECT_VERSION_SUFFIX_SHORT}"
   )
 
-SET(FN_PREFIX "FW${PROJECT_VERSION}${PROJECT_VERSION_SUFFIX_SHORT}")
+set(FN_PREFIX "FW${PROJECT_VERSION}${PROJECT_VERSION_SUFFIX_SHORT}")
 
-MESSAGE(WARNING     "
+message(
+  WARNING
+    "
 ***************** YOUR ATTENTION PLEASE *****************
 CMake support is experimental. There is no guarantee at this time. If you have problems you are encouraged to fall back to the tried-and-true methods.
 *********************** THANK YOU **********************
 We now return to your regularly scheduled Firmware Build."
-)
+  )
 
-OPTION(SECONDARY_LANGUAGES "Secondary language support in the firmware" ON)
+option(SECONDARY_LANGUAGES "Secondary language support in the firmware" ON)
 
-SET(MAIN_LANGUAGES cs de es fr it pl CACHE STRING "The list of 'main' languages to be included, in the correct order")
-SET(COMMUNITY_LANGUAGES nl ro hu hr sk sv no CACHE STRING "The list of community languages to be included, in the correct order")
-SET(SELECTED_LANGUAGES ${MAIN_LANGUAGES} ${COMMUNITY_LANGUAGES})
+# Language configuration
+set(MAIN_LANGUAGES
+    cs de es fr it pl
+    CACHE STRING "The list of 'main' languages to be included, in the correct order"
+    )
+set(COMMUNITY_LANGUAGES
+    nl
+    ro
+    hu
+    hr
+    sk
+    sv
+    no
+    CACHE STRING "The list of community languages to be included, in the correct order"
+    )
+set(SELECTED_LANGUAGES ${MAIN_LANGUAGES} ${COMMUNITY_LANGUAGES})
 
 get_dependency_directory(prusa3dboards PRUSA_BOARDS_DIR)
 project(Prusa-Firmware)
 add_subdirectory(lib)
 
-FILE(STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/Firmware/config.h MAX_SIZE_LINE REGEX "^#define \+LANG_SIZE_RESERVED \+")
-STRING(REGEX MATCH "0x[0-9]+" MAX_SIZE_HEX "${MAX_SIZE_LINE}")
+# Get LANG_MAX_SIZE from sources
+file(STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/Firmware/config.h MAX_SIZE_LINE
+     REGEX "^#define \+LANG_SIZE_RESERVED \+"
+     )
+string(REGEX MATCH "0x[0-9]+" MAX_SIZE_HEX "${MAX_SIZE_LINE}")
 math(EXPR LANG_MAX_SIZE "${MAX_SIZE_HEX}" OUTPUT_FORMAT DECIMAL)
 message("Language maximum size (from config.h): ${LANG_MAX_SIZE} bytes")
 
-set (LANG_BIN_MAX 249856) # Ditto, this in xflash_layout.h but needs invocation of the preprocessor... :-/
+set(LANG_BIN_MAX 249856) # Ditto, this in xflash_layout.h but needs invocation of the
+                         # preprocessor... :-/
```

Comment:
this is also garbage reformat,, why split the comment?

Reply:
cmake-format is bad, or at least the defaults we are using are pretty bad. I've set the same values as the Buddy FW for the moment and just accepted the garbage...

Reply:
Not sure I agree with that mindset, but :shrug:

Reply:
Following 4 projects with where this is being used, I'm just trying to make it uniform. I hate all autoformatters in general BTW.

If you can find some better settings for cmake-format (0.6.13, not the outdated 0.6.0 as used by buddy) I'm all ears. I'll try to push it though-out. I tried a bit in the past, but it always looked like garbage no matter what I tried.

Reply:
skimmed the docs and didn't see anything particularly useful configuration wise :(

Reply:
Looking at buddy, where there isn't much discipline, I came to appreciate autoformatters. But I'm with you here, cmake-format is just subpar in everything it does. Some flags help with one case and destroy 20 more. I'm just trying to jump a little bit ahead and clone what we do on MMU here.

I wish I had alternatives.

Reply:
change it back to tabs and I'll be happy /s :rofl:

Reply:
"ColumnLimit" setting for that one thing (?).

OctoPrint for uses pre-commit hook (and tool with the same name) for checking and reformatting (and this is also wired into github actions for all commits and pull requests). 

https://pre-commit.com/
https://github.com/OctoPrint/OctoPrint/blob/maintenance/.pre-commit-config.yaml
https://github.com/OctoPrint/OctoPrint/blob/master/.github/workflows/build.yml

There are various reformatting hooks supported, including simple things like trailing-whitespace or end-of-file-fixer
while also more complicated things. 

~Full list:
https://pre-commit.com/hooks.html

c++ ones are also there but can't tell if these will fit your needs.

Example, manual run, for python project:
```
(venv) pi@prusa-octopi:~/devel/OctoPrint-Sdwire $ pre-commit run --hook-stage manual --all-files
fix end of files.........................................................Passed
trim trailing whitespace.................................................Failed
- hook id: trailing-whitespace
- exit code: 1
- files were modified by this hook

Fixing octoprint_sdwire/__init__.py

check for case conflicts.................................................Passed
check json...........................................(no files to check)Skipped
check yaml...............................................................Passed
check toml...........................................(no files to check)Skipped
check for merge conflicts................................................Passed
fix python encoding pragma...............................................Passed
OctoPrint codemod: batch.................................................Passed
isort....................................................................Passed
black....................................................................Passed
flake8...................................................................Passed
prettier.................................................................Passed
eslint...................................................................Passed
```

Reply:
@arekm See what we do here https://github.com/prusa3d/Prusa-Firmware-Buddy/blob/master/.pre-commit-config.yaml

Historically though this project was always buildable with just Arduino which makes is very approachable, but is also why many things are clunky.